MLAPI/Serialization/BitWriter.cs

```diff
@@ -182,7 +182,19 @@ public void WriteObjectPacked(object value)
             }
             else if (value.GetType().IsEnum)
             {
-                WriteInt32Packed((int)value);
+                var underlyingType = System.Enum.GetUnderlyingType(value.GetType());
+
+                if (underlyingType == typeof(Int32))
+                    WriteInt32Packed((int)value);
+                else if (underlyingType == typeof(byte))
+                    WriteByte((byte)value);
+                else if (underlyingType == typeof(Int16))
+                    WriteInt16Packed((short)value);
+                else else if (underlyingType == typeof(Int64))
```

Comment:
This is a bug! I dont have the repo here, will try to commit a fix. Should have just been else-if.